File: src/main/java/dev/slne/discord/spring/annotation/DiscordCommandMeta.java
package dev.slne.discord.spring.annotation;

import dev.slne.discord.guild.permission.CommandPermission;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;
import org.springframework.stereotype.Component;

@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
@Component
public @interface DiscordCommandMeta {

  String name();

  String description();

  CommandPermission permission();

  boolean guildOnly() default true;

  boolean nsfw() default false;
}

File: src/main/kotlin/dev/slne/discord/annotation/DiscordCommandMeta.kt
package dev.slne.discord.annotation

import dev.slne.discord.guild.permission.CommandPermission
import org.springframework.stereotype.Component

@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
@Component
annotation class DiscordCommandMeta(
    val name: String,
    val description: String,
    val permission: CommandPermission,
    val guildOnly: Boolean = true,
    val nsfw: Boolean = false,
    val ephemeral: Boolean = true,
    val sendTyping: Boolean = false
)
